code/datums/components/full_auto_fire.dm

```diff
@@ -142,7 +142,7 @@
 	if(get_dist(source.mob, target) < 2) //Adjacent clicking.
 		return
 
-	if(isnull(location)) //Clicking on a screen object.
+	if(istype(target, /obj/screen)) //Clicking on a screen object.
```

Comment:
why

Reply:
Its what caused the problem with not being able to click radials on autofire, the isnull(location) doenst work and it runtimes, on autofire refactor bravmole does this as well